msteams-platform/concepts/deploy-and-publish/appsource/prepare/frequently-failed-cases.md

```diff
@@ -1,246 +1,348 @@
-* Please include a *Settings* tab that is available from the app header.
+### 5.4 Usability
 
-![Wide idea configuration page](../../../../assets/images/faq/wideidea.png)
+* Users must be able to undo their last action in the tab.
+* Personal app tabs may aggregate content from team or group chat scoped instances of that app in a single view.
+* Tabs must be responsive to Teams themes. When the Teams theme is changed, the theme within the app must also change to reflect that theme.
+* Personal app tabs must use Teams-styled components (adopting Teams fonts, type ramps, color palettes, grid system, motion, tone of voice, etc.) whenever possible.
+* Include a Settings tab.
+* Tabs must use Teams interaction behaviors (in-page navigation, position and use of dialogs, information hierarchies, etc.) whenever possible.
 
-### &#9989; Tab configuration must happen in the configuration screen
+## 6.0 Bots
 
-* The configuration screen must clearly explain the value of the experience and how to configure the tab.
-* The configuration process must always provide a way for the users to continue and not end the user experience. For example, do not show an empty board after the user has configured the tab.
-* The user sign-in process must be a part of the configuration process. Ensure to complete it in the Tab UI. After the user has completed the configuration and loaded the tab, no further action is required.
-* Don't show your entire webpage within the sign-in configuration pop-up window.
-* A user must always be able to finish the configuration experience, even if they can’t immediately find the content they’re looking for.
-* The configuration experience must provide options for the user to find their content, pin a URL, or create new content if it doesn’t exist.
-* The configuration experience must remain within the Teams context. The user shouldn’t have to leave the configuration experience to create content and then return to Teams to pin it.
-* Use the available viewport area efficiently. Do not waste it on using huge logos inside the configuration pop up.
+If your app includes a bot, following these guidelines can increase the likelihood your app will pass submission.
 
-![OneNote allows users to paste a OneNote link in case notes can not be found](../../../../assets/images/faq/tab-onenote-config.png)
+> [!TIP]
+> If you want information on creating a high-quality experience, see the [bot design guidelines](~/bots/design/bots.md).
+
+### 6.1 Bot commands
 
-![Users can always create new plan on planner in case there no existing ones](../../../../assets/images/faq/tab-planner-config.png)
+* All commands that your bot supports must work correctly, including the "Hi", "Hello", and "Help" command.
+* Supported bot commands must be listed in the app manifest.
+* Use command lists. Analyzing user input or predicting user intent is hard. Instead of letting users guess what your bot can do, provide a list of commands your bot understands.
```

Comment:
@heath-hamilton, Please update - 
•	Its highly recommended to list supported bot commands in app manifest. This helps user to converse efficiently with bot across all devices.  
•	To improve the bot user experience it is highly recommended to use Adaptive cards in bot responses

Reply:
@ramendhe, I think we need more detail as to why this is true--can you please elaborate?

"This helps user to converse efficiently with bot across all devices."

Reply:
@heath-hamilton We can update to include reason as - 
Its highly recommended to list supported bot commands in app manifest. This helps user to converse efficiently with bot across all devices, by choosing a command from prepopulated list instead of asking user to type it at each instance.